### DISCUSSION TITLE
float128 support

### DISCUSSION THREAD

**Discussion by fafour-lab** [pyccel/pyccel#1041] [Ideas]
**What I would like to be added**

I would like to see numpy's <code>float128</code> added to Pyccel's supported data types.
I need it for mathematical operations and to round it back to <code>float64</code> data type which is already supported by Pyccel.

**Language**

It would be great if it's supported by both C and Fortran

**Comment by EmilyBourne** [comment]
Blocked by #886

**Comment by jalalium** [comment]
the closest thing C has to float128 is long double, but unfortunately it is only 80 bit long and will not do the job since it loses in precision compared to float128. GCC has a nonstandard type __float128 but i don't think we can use that. Another alternative would be creating our own custom type from two doubles and reassigning the bits to match with the IEEE754 standard.
